// sidebarmenu/src/config/BasePath.js
export default {
  /* 首页 */
  H_: '',
  /* 公共 */
  FILE_ADDR: 'http://10.20.13.165/',
  /* 树 */
  ORGTREE_SELECTORGTREE: 'system/orgtree/selectOrgTree.do', // 组织机构树
  COMMODITYTREE_FACTORYBRAND: 'baseinfo/productbrand/selectTreeList.do', // 供应商品牌树。。。
  /* 下拉框 */
  SELECT_CODELIST_MAPLIST: 'system/codelist/mapList.do', // 参数分组。。
  SELECT_PARAMGRPCODEGROUP: 'system/orgrelationshiptypes/selectList.do', // 参数分组。。。
  SELECT_BRANDIDGROUP: 'baseinfo/productbrand/selectList.do', // 品牌。。。
  SELECT_PCKTMPIDGROUP: 'baseinfo/packagetemplate/selectList.do', // 包装模板
  SELECT_GOODSGRADEGROUP: 'baseinfo/productgrade/selectList.do', // 卷烟级别。。。
  SELECT_FACTORYIDGROUP: 'system/orgroles/selectList.do', // 厂家。。。
  SELECT_PROVINCEIDGROUP: 'system/regionalisms/selectList.do', // 省份。。。
  SELECT_ORGROLETYPIDGROUP: 'system/orgroletypes/selectList.do', // 组织角色。。。
  SELECT_ORGTYPEGROUP: 'system/orgtypes/selectList.do', // 组织机构类型。。。
  SELECT_DEPTIDGROUP: 'system/orgunits/selectList.do', // 所属部门。。。
  SELECT_RTLCATIDGROUP: 'system/customergroup/selectList.do', // 客户集ID。。。
  /* 基础信息 */
  /* 客户信息 */
  CUSTOMER_SELECT: '/system/customer/selectList.do', // 客户信息查询
  CUSTOMER_DELETE: '/system/customer/delete.do', // 客户删除
  CUSTOMER_SELECTONE: '/system/customer/selectOne.do', // 客户明细查询
  CUSTOMER_UPDATE: '/system/customer/update.do', // 客户信息更新
  /* 雇员信息 */
  EMPLOYEE_SELECT: 'system/employees/selectList.do', // 雇员信息查询
  EMPLOYEE_SELECTONE: 'system/employees/selectOne.do',
  EMPLOYEE_INSERT: 'system/employees/insert.do',
  EMPLOYEE_UPDATE: 'system/employees/update.do',
  EMPLOYEE_DELETE: 'system/employees/delete.do',
  EMPLOYEE_DELETEBATCH: 'system/employees/deleteBatch.do',
  /* 行政区 */
  REGIONALISMS_SELECT: 'system/regionalisms/selectList.do', // 行政区查询
  /* 银行 */
  BANK_SELECT: 'system/bank/selectList.do', // 银行查询
  /* 商品信息 */
  COMMODITY_SELECT: 'baseinfo/cigarettes/selectList.do', // ...
  COMMODITY_PACKAGEMX: 'baseinfo/goodspackage/selectList.do', // ...
  /* 客户集 */
  CUSTOMERGROUP_SELECTLIST: 'system/customergroup/selectList.do',
  CUSTOMERGROUP_INSERT: 'system/customergroup/insert.do',
  CUSTOMERLISTBYCUSTOMERGROUP: 'system/common/getCustomerGroupMembers.do', // 根据客户集id查客户
  CUSTOMERGROUP_DELETE: 'system/customergroup/delete.do',
  CUSTOMERGROUP_INSERTCUSTOMERS: 'system/customerclassified/insertBatch.do',
  /* 商品集 */
  CIGCATEGORYS_SELECTLIST: 'baseinfo/cigcategorys/selectList.do',
  CIGCATEGORYS_INSERT: 'baseinfo/cigcategorys/insert.do',
  /* 岗位 */
  DEPTPLACES_DELETE: 'system/deptplaces/delete.do',
  DEPTPLACES_SELECT: 'system/deptplaces/selectList.do',
  DEPTPLACES_UPDATE: 'system/deptplaces/update.do',
  DEPTPLACES_INSERT: 'system/deptplaces/insert.do',
  DEPTPLACES_DELETEBATCH: 'system/deptplaces/deleteBatch.do',
  /* 服务 */
  DEPTPLACESTYPE_DELETE: 'system/deptplacestype/delete.do',
  DEPTPLACESTYPE_SELECT: 'system/deptplacestype/selectList.do',
  DEPTPLACESTYPE_UPDATE: 'system/deptplacestype/update.do',
  DEPTPLACESTYPE_INSERT: 'system/deptplacestype/insert.do',
  DEPTPLACESTYPE_DELETEBATCH: 'system/deptplacestype/deleteBatch.do',
  /* 系统管理 */
  ROLE_SELECT: '/system/role/selectListPages.do', // 角色查询
  ROLE_INSERT: '/system/role/insert.do', // 角色新增
  ROLE_UPDATE: '/system/role/update.do', // 角色更新
  ROLE_DELETE: '/system/role/delete.do', // 角色删除
  ROLE_PERMISSIONS_QUERY: '/system/rolepermissions/selectList.do', // 角色赋权查询
  ROLE_PERMISSIONS_DELETE: '/system/rolepermissions/delete.do', // 角色赋权删除
  ROLE_USER_QUERY: 'system/userrole/selectList.do', // 角色用户授权查询
  ROLE_USER_DELETE: 'system/userrole/delete.do', // 角色用户授权删除
  ROLE_USER_JURISDICTION: 'system/common/selectUserByComAndRole.do', // 角色人员权限
  ROLE_USER_DISTRIBUTION: 'system/common/selectRolebyResources.do', // 角色模块分配
  ROLE_USER_INSERT_BATCH: '/system/userrole/insertBatch.do', // 角色人员权限批量插入
  ROLE_USER_DELETE_BATCH: '/system/userrole/deleteBatch.do', // 角色人员权限批量删除
  ROLE_PERMISSIONS_SELECT: 'system/common/selectRolebyResources.do', // 角色菜单赋权查询
  ROLE_PERMISSIONS_INSERT_BATCH: 'system/rolepermissions/insertBatch.do', // 角色菜单赋权批量插入
  ROLE_PERMISSIONS_DELETE_BATCH: 'system/rolepermissions/deleteBatch.do', // 角色菜单赋权批量插入
  /* 参数设置 */
  APPPARAM_DELETE: 'system/appparam/delete.do',
  APPPARAM_SELECT: 'system/appparam/selectList.do',
  APPPARAM_UPDATE: 'system/appparam/update.do',
  APPPARAM_INSERT: 'system/appparam/insert.do',
  APPPARAM_DELETEBATCH: 'system/appparam/deleteBatch.do',
  APPPARAMVALUE_SELECT: 'system/appparamvalue/selectList.do',
  APPPARAMVALUE_UPDATE: 'system/appparamvalue/update.do',
  APPPARAMVALUE_INSERT: 'system/appparamvalue/insert.do',
  APPPARAMVALUE_DELETE: 'system/appparamvalue/delete.do',
  /* 组织机构 */
  ORGUNITS_DELETE: 'system/orgunits/delete.do',
  ORGUNITS_INSERT: 'system/orgunits/insert.do',
  ORGUNITS_UPDATE: 'system/orgunits/update.do',
  ORGUNITS_SELECT: 'system/orgunits/selectList.do',
  /* 用户管理 */
  USER_INSERT: 'system/user/insert.do',
  USER_SELECTLISTPAGES: 'system/user/selectListPages.do',
  USER_SELECTLIST: 'system/user/selectList.do',
  USER_UPDATE: 'system/user/update.do',
  USER_DELETE: 'system/user/delete.do',
  /* 模块管理 */
  MODULE_SELECTLIST: 'system/resources/selectList.do',
  MODULE_INSERTIDTREE: '/system/resources/insertIdTree.do',
  MODULE_SELECTTREELIST: 'system/resources/selectTreeList.do',
  MODULE_DELETE: 'system/resources/delete.do',
  MDOULE_UPDARE: 'system/resources/update.do',
  /* 数据字典 */
  CODETYPE_SELECTLIST: 'system/codetype/selectList.do', // 查询主表列表
  CODETYPE_INSERT: 'system/codetype/insert.do', // 新增单条主表
  CODELIST_INSERT: 'system/codelist/insertBatch.do', // 批量新增、保存、删除从表对象
  CODETYPE_DELETE: 'system/codetype/delete.do', // 删除单条主表
  CODELIST_SELECT: 'system/codelist/selectList.do', // 根据主表id查询从表数据
  CODETYPE_UPDATE: 'system/codetype/update.do', // 更新单条主表
  /* 客戶拜訪 */
  /* 客户经理日历首页 */
  CLIENTMANAGER_ORDER_TOTAL: 'ordercdc/common/ordselectCust.do', // 未订单，未扣款，当日订单统计
  CLIENTMANAGER_CALENDAR: 'visit/common/selectCustmgrStatistics.do', // 客户经理日历接口
  /* 拜访计划制定 */
  VISITPLAN_SAVE: 'visit/visitingrec/insertBatch.do', // ...
  VISITPLAN_PLANDATE: 'visit/visitingrec/selectVisitPeriod.do', // ...
  VISITPLAN_SELECT: 'visit/visitingrec/selectMapListPages.do', // ...
  VISITPLAN_TOTAL: 'visit/visitingrec/selectVisitedCustomerSum.do', // ...
  /* 拜访计划制定 */
  VISITQUERY_SELECT: 'visit/visitingrec/selectVistingResult.do', // ...
  VISITQUERY_STEPSSELECT: 'visit/clsteps/selectList.do', // ...
  /* 采集计划制定 */
  JOBSET_SELECT: 'system/businesscalendar/selectGatherScheduleCustom.do', // ...
  JOBSET_UPDATE: 'system/businesscalendar/updateGatherScheduleCustom.do', // ...
  /* 监控地图 */
  MARKETMGR_STATISTICS: 'visit/common/selectMarketmgrStatistics.do', // 市场经理查看客户经理拜访情况
  CUSMGR_STATISTICS: 'visit/common/selectCustmgrStatistics.do', // 周期内拜访情况统计
  CUSTMGR_STATUS: 'visit/common/selectCustmgrStatus.do', // 周期内拜访情况统计
  /* 问卷调查 */
  FW_FROMS_INSERT: 'visit/forms/insert.do', // 插入表单
  FW_FROMS_SELECTLIST: 'visit/forms/selectList.do', // 查询所有问卷
  FW_ITEMS_SELECTELIST: 'visit/items/selectList.do',
  FW_FROMS_UPDATE: 'visit/forms/update.do', // 逻辑删除某问卷
  FW_ITEMS_GETFULLFORM: 'visit/forms/getFullForm.do', // 查詢问卷
  FW_ITEMS_SAVEFULLFORM: 'visit/forms/saveFullForm.do', // 保存问卷
  CL_STEPS: 'visit/clsteps/insertMuti.do', // 问卷录入
  /* 消费营销-活动管理 */
  ACTIVITY_CIGARETTES_NEW: '/visit/cigarettesnew/selectList.do', // ...
  ACTIVITY_SELECTCRMMATTERSLIST: 'visit/common/selectCrmMattersList.do', // 3.5.2消费营销-活动管理-活动管理列表
  ACTIVITY_INSERT: 'visit/plans/insert.do', // 3.5.3消费营销-活动管理-活动管理列表-新增（初始化及保存）
  ACTIVITY_UPDATE: 'visit/plans/update.do', // 3.5.4消费营销-活动管理-活动管理列表-修改（初始化）
  ACTIVITY_DELETE: 'visit/plans/delete.do', // 3.5.5消费营销-活动管理-活动管理列表-删除
  ACTIVITY_SELECTONE: 'visit/plans/selectOne.do',
  SELECT_MATTERIDGROUP: 'visit/crmmatters/selectList.do', // 3.5.7消费营销-活动管理-活动管理列表-活动事项下拉框
  ACTIVITY_SELECTACTIVITYCOLLECTION: 'visit/common/selectActivityCollection.do', // 3.5.8消费营销-活动管理-活动管理列表-新增-收集内容列表
  ACTIVITY_CUSTOMER_SELECT: 'system/common/custSelect.do', // 3.5.7消费营销-活动管理-活动管理列表-活动事项下拉框
  ACTIVITY_SELECTLIST: 'visit/plans/selectList.do',
  /* 历史拜访 */
  VISI_HISTORY: '/visit/visitingrec/selectList.do',
  /* 历史订单 */
  ORDER_HISTORY: 'orderquery/common/ordhisselect.do',
  ORDER_HISTORY_DETAIL: 'orderquery/common/ordhisselectLine.do',
  /* 组织机构树 */
  ORGUNIT_TREE: '/system/orgunits/selectTreeList.do',
  EMPLOYEE_TREE: '/system/employees/selectTreeList.do',
  /* 价格采集录入 */
  GATHER_PRICE_SELECT: '/visit/gatherproductprice/selectList.do',
  GATHER_PRICE_DETAIL: '/visit/gatherproductpricedetail/selectList.do', // 价格采集录入细表查询
  GATHER_PRICE_ADD: '/visit/gatherproductpricedetail/selectGatherPriceCigarettes.do', // 价格采集录入新建/修改
  GATHER_PRICE_UPDATE: '/visit/gatherproductpricedetail/saveGatherPrice.do',
  GATHER_PRICE_SUBMIT: '/visit/gatherproductprice/update.do',
  GATHER_PRICE_DELETE: '/visit/gatherproductprice/deleteGatherPriceAndDetail.do',
  /* 采集品规设置 */
  GATHER_SELECT: '/visit/gatherproduct/selectList.do', // 查询多条信息
  GATHER_SELECTONE: '/visit/gatherproduct/selectOne.do', // 查询单条信息
  GATHER_SELECTGATCH: '/visit/gatherproduct/selectGatherSettingCigarettes.do', // 价格采集商品设置
  GATHER_UPDATE: '/visit/gatherproduct/update.do', // 更新单条
  GATHER_UPDATEBATCH: '/visit/gatherproduct/updateBatch.do', // 更新多条
  GATHER_INSERT: '/visit/gatherproduct/insert.do', // 插入单条
  GATHER_INSERTBATCH: '/visit/gatherproduct/insertBatch.do', // 插入多条
  GATHER_DELETE: '/visit/gatherproduct/delete.do', // 删除单条记录
  GATHER_DELETEBATCH: '/visit/gatherproduct/deleteBatch.do', // 删除多条记录
  /* 3.6.1周订单分析报告 */
  WEEKORDERREPORT_SELECT: '/visit/reports/selectList.do', // 3.6.1.1接口一（订单报告查询）
  WEEKORDERREPORT_SELECTONE: '/visit/reports/selectOne.do', // 3.6.1.1接口一（订单报告查询）
  WEEKORDERREPORT_UPDATE: '/visit/reports/update.do', // 3.6.1.2接口二（保存）
  WEEKORDERREPORT_INSERT: '/visit/reports/insert.do', // 3.6.1.2接口二（保存）
  WEEKORDERREPORT_DELETE: '/visit/reports/delete.do', // 3.6.1.2接口二（删除）
  WEEKORDERREPORT_CONSUMPTION: '/baseinfo/common/selectKeyBrandMarketPrice.do', // 3.6.1.3接口三（订单报告参考数据1:重点品牌市场价格）
  WEEKORDERREPORT_BRANDS: '/baseinfo/common/selectKeyBrandRate.do', // 3.6.1.4接口四（订单报告参考数据2:各种率分析）
  WEEKORDERREPORT_CUSTOMER: '/baseinfo/common/selectKeyBrandIndex.do', // 3.6.1.5接口五（订单报告参考数据3:辖区指标分析）
  WEEKORDERREPORT_SELECTLIST: '/visit/common/selectWeekReportByObjId.do',
  WEEK_SELECTLIST: '/orderquery/dtweek/selectList.do',
// WEEKORDERREPORT_BRANDQUERY: '', // 3.6.1.6接口六（选择分析品牌）
  // WEEKORDERREPORT_BRANDSAVE: '', // 3.6.1.7接口七（选择分析品牌保存）
  /* 客户销售分析报告 */
  CUSTSALESANALYSIS_SELECT: '/visit/common/reportQuery.do', // 客户销售分析查询
  CUSTSALESANALYSIS_ANAYSIS: '/visit/common/reportQuery.do', // 月度总体卷烟经营情况
  CUSTSALESANALYSIS_UPDATE: '/visit/reports/update.do', // 3.6.1.2接口二（保存）
  CUSTSALESANALYSIS_INSERT: '/visit/reports/insert.do', // 3.6.1.2接口二（保存）
  CUSTSALESANALYSIS_SELECTONE: '/visit/reports/selectOne.do', // 3.6.1.2接口二（保存）
  CUSTSALESANALYSIS_ANAYSIS4: '/orderquery/common/reportQueryRefer4.do', // 3.6.2.3 分析报告-客户销售分析报告-客户分析报告参考数据4:总体指标分析
  CUSTSALESANALYSIS_ANAYSIS5: '/orderquery/common/reportQueryRefer5.do', // 3.6.2.4 分析报告-客户销售分析报告-客户分析报告参考数据5:客户经理总体指标分析
  CUSTSALESANALYSIS_ANAYSIS6: '/orderquery/common/reportQueryRefer6.do', // 3.6.2.5 分析报告-客户销售分析报告-客户分析报告参考数据6规格前5销量
  CUSTSALESANALYSIS_ANAYSIS7: '/orderquery/common/reportQueryRefer7.do', // 3.6.2.6 分析报告-客户销售分析报告-客户分析报告参考数据7分价位订购情况
  WEEKORDERREPORT_WEEK: '/orderquery/common/selectReportPeriod.do', // 3.6.1.7接口七（选择分析品牌保存）
  /* 3.6.1周订单分析报告 */
  REGIONALSALES_ANALYTICALDATA1_KEY: '/orderquery/common/areaReportQueryRefer8.do', // 3.6.3.3接口三（区域销售分析参考数据8: 关键指标）
  REGIONALSALES_ANALYTICALDATA2_CATEGORY: '/orderquery/common/areaReportQueryRefer9.do', // 3.6.3.4接口四（区域销售分析参考数据9: 五类烟指标）
  REGIONALSALES_ANALYTICALDATA3_PRICE: '/orderquery/common/areaReportQueryRefer10.do', // 3.6.3.5接口五（区域销售分析参考数据10: 分价类）
  REGIONALSALES_ANALYTICALDATA4_NEWPRODUCTS: '/orderquery/common/areaReportQueryRefer11.do', // 3.6.3.6***接口六（区域销售分析参考数据11: 新品分析）
  REGIONALSALES_ANALYTICALDATA5_SALES: '/orderquery/common/areaReportQueryRefer12.do', // 3.6.3.7接口七（区域销售分析参考数据12: 客户档次变化情况及销售分析）
  REGIONALSALES_QUARTER: '/visit/bizperiods/selectList.do', // 区域销售分析参考数据--季度
  REGIONALSALES_QUARTER_SELECTLIST: '/orderquery/dtquarter/selectOne.do', // 区域销售分析参考数据--季度
  /* 投诉记录 */
  COMPLAINTREC_SELECTNAME: '/visit/complaintrec/selectComplaintRecDepName.do', // 查询带部门名称的投诉记录
  COMPLAINTREC_SELECT: 'visit/complaintrec/selectList.do', // 查询多条
  COMPLAINTREC_SELECTONE: '/visit/complaintrec/selectOne.do', // 查询单条
  COMPLAINTREC_UPDATE: '/visit/complaintrec/update.do', // 更新单条记录
  COMPLAINTREC_UPDATEBATCH: '//visit/complaintrec/updateBatch.do', // 更新多条记录
  COMPLAINTREC_INSERT: '/visit/complaintrec/insert.do', // 插入单条记录
  COMPLAINTREC_INSERTBATCH: '//visit/complaintrec/insertBatch.do', // 插入多条记录
  COMPLAINTREC_DELETE: '/visit/complaintrec/delete.do', // 删除单条记录
  COMPLAINTREC_DELETEBATCH: '/visit/complaintrec/deleteBatch.do' // 删除多条记录
}
