// sidebarmenu/build/webpack.base.conf.js
var webpack = require('webpack')
var path = require('path')
var utils = require('./utils')
var config = require('../config')
var vueLoaderConfig = require('./vue-loader.conf')

function resolve (dir) {
  return path.join(__dirname, '..', dir)
}

module.exports = {
  entry: {
    app: './src/main.js'
  },
  output: {
    path: config.build.assetsRoot,
    filename: '[name].js',
    // sourceMapFilename: "[name].js.map",
    publicPath: process.env.NODE_ENV === 'production'
      ? config.build.assetsPublicPath
      : config.dev.assetsPublicPath,
    library: 'SidebarMenu',
    libraryTarget: 'umd',
    umdNamedDefine: true
  },
  externals: {
    vue: {
      root: 'Vue',
      commonjs: 'vue',
      commonjs2: 'vue',
      amd: 'vue'
    }
  },
  resolve: {
    extensions: ['.js', '.vue', '.json'],
    alias: {
      '@': resolve('src'),
      'static': path.resolve(__dirname, '../static'),
      'jquery': path.resolve(__dirname, '../static/js/plugins/jQuery/jQuery-2.2.0.min'),
      'vendor': path.resolve(__dirname, '../src/utils'),
      'Information': path.resolve(__dirname, '../src/components/Information'),
      'Visit': path.resolve(__dirname, '../src/components/Visit'),
      'System': path.resolve(__dirname, '../src/components/System'),
      'Acquisition': path.resolve(__dirname, '../src/components/Acquisition'),
      'Collection': path.resolve(__dirname, '../src/components/Collection'),
      'BrandActivity': path.resolve(__dirname, '../src/components/BrandActivity'),
      'Report': path.resolve(__dirname, '../src/components/Report'),
      'Template': path.resolve(__dirname, '../src/components/Template'),
    }
  },
  // 增加一个plugins
  plugins: [
    new webpack.ProvidePlugin({
      $: "jquery",
      jQuery: "jquery"
    })
  ],
  module: {
    rules: [
      {
        test: /\.(js|vue)$/,
        loader: 'eslint-loader',
        enforce: 'pre',
        include: [resolve('src'), resolve('test')],
        options: {
          formatter: require('eslint-friendly-formatter')
        }
      },
      {
        test: /\.vue$/,
        loader: 'vue-loader',
        options: vueLoaderConfig
      },
      {
        test: /\.js$/,
        loader: 'babel-loader',
        include: [resolve('src'), resolve('test')]
      },
      // {
      //   test:/\.(css|scss)$/,
      //   include: /node_modules/,
      //   loader: ['style-loader', 'css-loader', 'postcss-loader', 'sass-loader']
      // },
      {
        test: /\.(png|jpe?g|gif|svg)(\?.*)?$/,
        loader: 'url-loader',
        options: {
          limit: 10000,
          name: utils.assetsPath('img/[name].[hash:7].[ext]')
        }
      },
      {
        test: /\.(mp4|webm|ogg|mp3|wav|flac|aac)(\?.*)?$/,
        loader: 'url-loader',
        options: {
          limit: 10000,
          name: utils.assetsPath('media/[name].[hash:7].[ext]')
        }
      },
      {
        test: /\.(woff2?|eot|ttf|otf)(\?.*)?$/,
        loader: 'url-loader',
        options: {
          limit: 10000,
          name: utils.assetsPath('fonts/[name].[hash:7].[ext]')
        }
      }
    ]
  },
}
